### DISCUSSION TITLE
Language is not showing up

### DISCUSSION THREAD

**Discussion by Purfview** [github-linguist/linguist#6236] [Classification] [answered]
I have a JavaScript repo where the language is not being detected: https://github.com/Purfview/Discogs-Scout
Other JavaScript repos I have do show that as the repo language.

### Expected behaviour
 
Repo gets "JavaScript" tag.

**Comment by lildude** [comment, marked as answer]
This is happening because linguist is detecting the file as being minified JavaScript due to many many lines that are [longer then 110 chars and is thus considering it generated](https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/lib/linguist/generated.rb#L152-L166) and this ignores the file.

```console
$ github-linguist Discogs-Scout/Discogs_Scout.user.js
Discogs-Scout/Discogs_Scout.user.js: 792 lines (734 sloc)
  type:      Text
  mime type: application/javascript
  language:  JavaScript
  appears to be generated source code
$
``` 

You'll need to implement an [override to explicitly mark the file as not being a generated file](https://github.com/github/linguist/blob/master/docs/overrides.md#generated-code).
